### ISSUE TITLE
Can we export SAM in tflite?

### ISSUE THREAD

**Issue by danishbansal-unthinkable** [facebookresearch/segment-anything#367]
Any idea if we can export the SAM model in Tflite so that we can integrate it on edge in Android/ios? If yes can anyone help by providing the code to achieve it?

**Comment by kbrandwijk** [comment]
Apparently, it's been tried here, with some limitations/workarounds: https://github.com/csia-pme/djl-image-sam/tree/main/pytorch_convert#modifypatch-the-model

**Comment by ryouchinsa** [comment]
Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)

**Comment by ryouchinsa** [comment]
Improved "Create polygon using SAM" feature so that you can label pixels using the pixels option. You can label [Segment Anything 1 Billion (SA-1B)](https://ai.meta.com/datasets/segment-anything-downloads/) like dataset by yourself.

![sam-pixels](https://github.com/facebookresearch/segment-anything/assets/1954306/d828c18b-6fda-4258-b866-21bc34a0d17f)
